src/api/environment.cc

```diff
@@ -885,7 +885,8 @@ void AddLinkedBinding(Environment* env,
             exports,
             module,
             context,
-            reinterpret_cast<napi_addon_register_func>(priv));
+            reinterpret_cast<napi_addon_register_func>(priv),
+            NODE_API_DEFAULT_MODULE_API_VERSION);
```

Comment:
Can we add an optional parameter so that the target version can be set with `AddLinkedBinding` since registering with `napi_addon_register_func` is the preferred approach to register a linked node-api addon?

Reply:
Good suggestion! The code is changed. It is now a bit more complex than before, but it works. There are new unit tests to verify it.